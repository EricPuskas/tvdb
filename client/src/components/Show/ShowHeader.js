import React from "react";
import Rating from "react-rating";

const ShowHeader = ({ title, genres, rating, ratingText }) => {
  return (
    <div className="Show-title">
      <h1> {title} </h1>
      <ul className="tags">{genres}</ul>
      <div>
        <Rating
          emptySymbol="far fa-star"
          fullSymbol="fas fa-star"
          stop={10}
          initialRating={rating}
          readonly
        />{" "}
        <span style={{ display: "block" }}>{ratingText}</span>
      </div>
    </div>
  );
};

export default ShowHeader;
