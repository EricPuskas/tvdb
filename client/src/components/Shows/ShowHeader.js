import React from "react";
import Moment from "react-moment";
import isValidDate from "../../utils/isValidDate";

const ShowHeader = ({ title, firstAired, network }) => {
  return (
    <>
      <h2> {title} </h2>
      <hr />
      <div>
        First Aired:{" "}
        <strong>
          {isValidDate(firstAired) ? (
            <Moment format="MMM DD (YYYY)">{firstAired}</Moment>
          ) : (
            "Unknown"
          )}
        </strong>
      </div>
      <div>
        Network: <strong>{network ? network : "Unknown"}</strong>
      </div>
      <hr />
    </>
  );
};

export default ShowHeader;
