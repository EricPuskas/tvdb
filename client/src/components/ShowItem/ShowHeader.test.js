import React from "react";
import { shallow } from "enzyme";
import ShowHeader from "./ShowHeader";

describe("Show Header Component", () => {
  const warn = console.warn;
  console.warn = jest.fn();
  describe("Invalid data for Network and Date", () => {
    const propsInvalid = {
      title: "Arrow",
      firstAired: "not_a_valid_date",
      network: ""
    };
    const component = shallow(<ShowHeader {...propsInvalid} />);
    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should display 'Unknown' where data is invalid", () => {
      const date = component
        .find(`[data-test="firstAired"]`)
        .childAt(0)
        .debug();
      const network = component
        .find(`[data-test="network"]`)
        .childAt(0)
        .debug();
      expect(date).toEqual("Unknown");
      expect(network).toEqual("Unknown");
    });

    expect(console.warn).toHaveBeenCalled();
  });

  describe("Valid data for Network and Date", () => {
    const propsValid = {
      title: "Arrow",
      firstAired: "2017-03-31",
      network: "HBO"
    };
    const component = shallow(<ShowHeader {...propsValid} />);
    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should display accordingly", () => {
      const date = component.find(`[data-test="firstAired"]`).props();
      const network = component
        .find(`[data-test="network"]`)
        .childAt(0)
        .debug();

      expect(date.children.props.children).toEqual(propsValid.firstAired);
      expect(network).toEqual(propsValid.network);
    });
  });
  console.warn = warn; // Return things to normal so other tests aren't affected.
});
