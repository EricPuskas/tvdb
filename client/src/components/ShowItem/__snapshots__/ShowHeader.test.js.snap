// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Header Component Invalid data for Network and Date Should pass snapshot test 1`] = `"<h2> Arrow </h2><hr/><div>First Aired: <strong data-test=\\"firstAired\\">Unknown</strong></div><div>Network: <strong data-test=\\"network\\">Unknown</strong></div><hr/>"`;

exports[`Show Header Component Valid data for Network and Date Should pass snapshot test 1`] = `"<h2> Arrow </h2><hr/><div>First Aired: <strong data-test=\\"firstAired\\"><time dateTime=\\"1490907600000\\">Mar 31 (2017)</time></strong></div><div>Network: <strong data-test=\\"network\\">HBO</strong></div><hr/>"`;
